ansible_collections/arista/avd/roles/eos_validate_state/python_modules/tests/avdtestsecurity.py

```diff
@@ -36,3 +36,45 @@ def test_definition(self) -> dict | None:
         anta_tests.append({"VerifyAPIHttpsSSL": {"profile": profile, "result_overwrite": {"custom_field": f"eAPI HTTPS SSL Profile: {profile}"}}})
 
         return {self.anta_module: anta_tests}
+
+
+class AvdTestIPSecurity(AvdTestBase):
+    """
+    AvdTestIPSecurity class for IP security connection tests.
+    """
+
+    anta_module = "anta.tests.security"
+
+    @cached_property
+    def test_definition(self) -> dict | None:
+        """
+        Generates the proper ANTA test definition for all IP security connection tests.
+
+        Returns:
+            test_definition (dict): ANTA test definition.
+        """
+        anta_tests = []
+
+        # Check if there are any path groups with static peers
+        if (path_groups := get(self.structured_config, "router_path_selection.path_groups")) is None:
+            LOGGER.info("No router path-group configured to collect the static peer. %s is skipped.", self.__class__.__name__)
+            return None
+
+        added_peers = set()
+        for path_group in path_groups:
+            static_peers = path_group.get("static_peers", [])
+            for peer in static_peers:
+                peer_address = peer.get("router_ip")
+                vrf = "default"  # TODO: Keeping the vrf name static for now. We may need to change later on.
```

Comment:
Not very familiar with the WAN data models but is there a VRF key in the structured config?

Reply:
no its not right now thats why keeping default for now